temporal-sdk/src/main/java/io/temporal/internal/worker/TrackingSlotSupplier.java

```diff
@@ -64,7 +64,8 @@ public Optional<SlotPermit> tryReserveSlot(SlotReservationData dat) {
 
   public void markSlotUsed(SI slotInfo, SlotPermit permit) {
     if (permit == null) {
-      return;
+      throw new IllegalStateException(
```

Comment:
Nit: `IllegalArgumentException`

Reply:
Also doesn't this same check also apply to release as well?

Reply:
I picked state because a user seeing this never specified the argument and doesn't care. It's illegal state in the sense that the SDK is in an illegal state. Yeah, will add to release too.

Reply:
So `IllegalStateException` is if a method is invoked at an inappropriate time, `IllegalArgumentException` is if the argument was invalid, here the argument was invalid

Reply:
Like I understand your point, but these exceptions have pretty standard meanings and I'd prefer to stick to them.

Reply:
Yeah I just think it's very debatable because the user _has no control_ over the argument. I'll change it since it's really not a big deal either way.